src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java

```diff
@@ -69,14 +68,27 @@ public JsonResult execute() {
         String entityType = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
 
         List<FeedbackSessionAttributes> feedbackSessionAttributes;
+        Map<String, Map<String, String>> courseIdAndSessionNameToStudent = new HashMap<>();
         List<InstructorAttributes> instructors = new ArrayList<>();
 
         if (courseId == null) {
             if (entityType.equals(Const.EntityType.STUDENT)) {
                 List<StudentAttributes> students = logic.getStudentsForGoogleId(userInfo.getId());
                 feedbackSessionAttributes = new ArrayList<>();
                 for (StudentAttributes student : students) {
-                    feedbackSessionAttributes.addAll(logic.getFeedbackSessionsForCourse(student.getCourse()));
+                    String studentCourseId = student.getCourse();
+                    String emailAddress = student.getEmail();
+                    List<FeedbackSessionAttributes> sessions = logic.getFeedbackSessionsForCourse(studentCourseId)
+                            .stream()
+                            .map(session -> session.sanitizeForStudent(emailAddress))
+                            .collect(Collectors.toList());
+                    feedbackSessionAttributes.addAll(sessions);
+                    if (!courseIdAndSessionNameToStudent.containsKey(studentCourseId)) {
+                        courseIdAndSessionNameToStudent.put(studentCourseId, new HashMap<>());
+                    }
```

Comment:
I don't believe we can make the assumption that `courseId` is unique either. The only guarantee is that `email` + `courseId` is unique, as this is the key of the entity. There is no guarantee that `account` + `courseId` is unique.

Reply:
Just to clarify, is your concern that a student may have multiple email addresses within the same course? That's true, but isn't this method only called when the student is logged in? Wouldn't they only have one email address just within this context then, which is the email address tied to the account? Otherwise, if not, if accounts can be tied to multiple email addresses, then within any feedback session in any course, there is no way for me to tell for which student email address the deadline should be adjusted for. Unless maybe the specific email address is passed as a parameter or something?

Reply:
One account can be tied to multiple student objects with different emails.

Reply:
While unlikely, there is no check to ensure that the email associated with the account is the email of the student. The only guarantee is that `courseId` + `email` is unique for each student.

Reply:
Thanks for clarifying. Then still, how do I know which of the student's email addresses within the course to use for adjusting deadlines and filtering? Because if a student can have multiple email addresses in a course, then all of them would be in the session together, but the deadline for a session can only be adjusted for one of the student's email addresses.

Reply:
The solution I was thinking of was to handle this at the source. I'll see if there's a cleaner solution and update.